create-rust-app/src/lib.rs

```diff
@@ -67,6 +70,17 @@ pub struct AppData {
     pub storage: Storage,
 }
 
+#[cfg(feature = "plugin_auth")]
+impl AppData {
+    pub fn with_custom_email_templates<T: EmailTemplates + 'static>(
+        mut self,
+        templates: T,
+    ) -> Self {
+        self.mailer = Mailer::new(Box::new(templates));
+        self
+    }
+}
+
```

Comment:
builder-esque function that lets users define their own custom email templates, or to use the default templates with a custom base_url